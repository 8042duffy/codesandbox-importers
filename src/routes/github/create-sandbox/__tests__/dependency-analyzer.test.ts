import getDependencies from '../dependency-analyzer';

const CODE = `
import React from "react"
import ReactDOM from "react-dom"
import { Provider } from "mobx-react"
import { observable, reaction } from "mobx"
import { onSnapshot, onAction, onPatch, applySnapshot, applyAction, applyPatch, getSnapshot } from "mobx-state-tree"
import "todomvc-app-css/index.css"
importoijaa from 'dawpoefk';
import createRouter from "./utils/router"
import App from "./components/App"
import SingleQuote from 'singlquote'
import "./index.css"

import { ShopStore } from "./stores/ShopStore"

require('test')

connectReduxDevtools(require("remotedev"), store)
`;

describe('dependency-analyzer', () => {
  it('can detect dependencies in the code', () => {
    const file = {
      title: '',
      code: CODE,
    };
    expect(getDependencies([file])).toMatchSnapshot();
  });

  it('can detect from multiple files', () => {
    const file = {
      title: '',
      code: CODE,
    };
    expect(getDependencies([file, file])).toMatchSnapshot();
  });
});
