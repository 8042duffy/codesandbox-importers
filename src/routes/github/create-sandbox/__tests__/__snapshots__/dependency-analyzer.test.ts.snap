// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dependency-analyzer can detect dependencies in the code 1`] = `
Array [
  "react",
  "react-dom",
  "mobx-react",
  "mobx",
  "mobx-state-tree",
  "todomvc-app-css/index.css",
  "singlquote",
  "test",
  "remotedev",
]
`;

exports[`dependency-analyzer can detect from multiple files 1`] = `
Array [
  "react",
  "react-dom",
  "mobx-react",
  "mobx",
  "mobx-state-tree",
  "todomvc-app-css/index.css",
  "singlquote",
  "test",
  "remotedev",
  "react",
  "react-dom",
  "mobx-react",
  "mobx",
  "mobx-state-tree",
  "todomvc-app-css/index.css",
  "singlquote",
  "test",
  "remotedev",
]
`;
